src/jit/valuenum.cpp

```diff
@@ -1188,7 +1265,7 @@ ValueNum ValueNumStore::VNForMapSelectWork(ValueNumKind vnk, var_types typ, Valu
     VNDefFunc2Arg fstruct(VNF_MapSelect, arg0VN, arg1VN);
     if (GetVNFunc2Map()->Lookup(fstruct, &res))
     {
-        return res;
+        goto Return;
```

Comment:
I strongly disagree, and I don't think that it should be necessary to put into our coding conventions that we should follow structured programming principles when it doesn't adversely impact the code (which, IMO, is nearly always the case). In this case, I don't think that turning these "if"s into "else if"s would "make the code much less readable".